### ISSUE TITLE
模型 – 实现 XGBoost v1 模型训练

### ISSUE THREAD

**Issue by phead198708** [phead198708/BiGan#17]
**目标**
训练第一版候选模型，输出概率分数。

**任务清单**
- [ ] 定义 XGBoost 参数空间
- [ ] 训练模型并在验证集调参
- [ ] 输出 feature importance
- [ ] 保存模型 artifact 与配置文件

**依赖**
- [ ] 模型 – 实现训练数据集装配脚本

**验收标准**
- [ ] XGBoost 模型可训练并保存
- [ ] 输出验证集指标
- [ ] 输出特征重要性结果

**Comment by phead198708** [comment]
Implemented issue #17 in commit `90a12cc`.

Summary:
- Added deterministic `xgboost-v1` candidate training under `bigan.modeling.xgboost_v1`.
- Because the project environment currently has no `xgboost`, `sklearn`, or `numpy` dependency, this v1 uses a dependency-free logistic-loss gradient boosted stump implementation with XGBoost-style gradient/hessian split gain.
- Defines a parameter search space for boosting rounds, learning rate, L2 penalty, and fixed v1 stump depth.
- Tunes candidates on validation Brier score, then emits train/val/test metrics.
- Saves reusable artifacts: `model.json`, `xgboost_config.json`, `metrics.json`, `feature_importance.json`, and `manifest.json`.
- Provides `load_xgboost_v1_model(...)`, probability prediction, and top feature contribution helpers for later calibration/prediction work.
- Added `bigan-ingest xgboost-v1` CLI command.

Verification:
- `.venv/bin/pytest tests/modeling/test_xgboost_v1.py -q`
- `.venv/bin/pytest tests/modeling -q`
- `.venv/bin/pytest tests/modeling tests/backtest -q`
- `.venv/bin/pytest -q` (passes; 2 live integration tests skipped)
- `.venv/bin/ruff check .`

**Comment by phead198708** [comment]
Verified in 90a12cc. `train_xgboost_v1` enumerates a `rounds × learning_rate × l2 × depth` grid, tunes on validation Brier (with ROC AUC tiebreak), persists `model.json`, `xgboost_config.json`, `metrics.json`, `feature_importance.json`, and `manifest.json`, and exposes `load_xgboost_v1_model`, `predict_proba_many`, and `top_feature_contributions`. Tests in `tests/modeling/test_xgboost_v1.py` cover tuning + artifact persistence + reproducibility, prediction explanation, and empty-grid rejection. CLI: `bigan-ingest xgboost-v1`.

The acceptance criteria are technically met, **but I recommend keeping this issue open** until one substantive deviation is acknowledged or resolved.

## Substantive deviation: the "XGBoost-v1" trainer is not actually XGBoost

The implementation is a hand-rolled depth-1 logistic-loss boosted-stump trainer using XGBoost-style gradient/hessian split-gain formulas. The closing comment is honest about this, but the implications are non-trivial:

```@/Users/tcscoder/Workspaces/BiGan/src/bigan/modeling/xgboost_v1.py:49-50
        if any(depth != 1 for depth in self.max_depth_grid):
            raise ValueError("xgboost-v1 currently supports max_depth=1 boosted stumps")
```

- The advertised `max_depth_grid` parameter is **hard-rejected for any value other than 1**, so the actual search space is just `rounds × learning_rate × l2_penalty` over depth-1 stumps. Stumps are a valid weak learner, but they are nowhere near a typical XGBoost configuration and cannot capture feature interactions.
- The persisted `model.json` is a **custom schema** (`algorithm: "logistic-loss-gradient-boosted-stumps"`), not portable to the xgboost native runtime / Spark / SageMaker / etc.
- The project has no `xgboost`, `numpy`, or `sklearn` dependency, so the choice is presumably driven by avoiding the wheel size. Adding xgboost would unblock real tree depth, regularization, monotonic constraints, GPU, and a portable artifact format.
- Tests assert `roc_auc == 1.0` on a tiny linearly-separable fixture — a stump can fit that perfectly, so the unit tests do not exercise a regime where stumps would underfit relative to a real GBDT.

## Suggested next step

Pick one and close this issue against it:

1. **Rename + scope**: Rename the artifact and module to `boosted-stump-v1` to make the engine choice explicit, then file a follow-up issue to add a real XGBoost trainer before any candidate is promoted to production.
2. **Adopt real XGBoost now**: Add `xgboost` to project dependencies, implement `train_xgboost_v1` against the real engine, persist the native booster JSON, and keep this current implementation as `BoostedStumpBaseline` if useful for offline-only scenarios.

Either path is fine, but the issue title says "实现 XGBoost v1 模型训练" and the artifact's `model_version` advertises `xgboost-v1`, so a downstream reader has no way to know the engine is actually a stump model unless they read the source.

Promotion (#19) intentionally compares the candidate against the logistic baseline using AUC + Brier deltas; on a real prediction-market dataset I would expect the stump model to be very close to logistic regression in margin. Worth keeping in mind before the first real promotion run.

**Comment by phead198708** [comment]
Addressed the review comment in commit `514d7aa`.

What changed:
- Replaced the custom dependency-free boosted-stump implementation with the native `xgboost` engine.
- Added project dependencies for `xgboost` and `numpy`.
- `train_xgboost_v1` now trains real `xgboost.Booster` models with a grid over rounds, learning rate, L2 regularization, max depth, gamma/min split loss, subsample, and colsample.
- `model.json` is now a native XGBoost booster JSON artifact, with `model_version`, `feature_columns`, and selected params stored as booster attributes.
- `load_xgboost_v1_model` loads the native booster artifact directly.
- Feature importance now comes from native XGBoost importance scores, and `top_feature_contributions` uses native `pred_contribs=True` contributions.
- Removed the exported custom `XGBoostV1Stump` type and updated prediction-contract tests accordingly.

Verification:
- `.venv/bin/pytest tests/modeling/test_xgboost_v1.py tests/modeling/test_predictions.py tests/modeling/test_calibration.py -q`
- `.venv/bin/pytest tests/modeling tests/backtest -q`
- `.venv/bin/pytest -q` (passes; 2 live integration tests skipped)
- `.venv/bin/ruff check .`

Local environment note: on macOS, native XGBoost required OpenMP runtime, so I installed `libomp` via Homebrew for local verification. This is the standard runtime prerequisite for the macOS XGBoost wheel.

This resolves the substantive deviation called out in the prior comment, so I am closing #17 as completed.
